### DISCUSSION TITLE
Controlling the random seed in train_test_split_edges for reproducibility

### DISCUSSION THREAD

**Discussion by sbonner0** [pyg-team/pytorch_geometric#2561] [Q&A]
Is it possible to control the seed used for `train_test_split_edges` to consistently return the same dataset split? Conversely, would it also be possible to set the seed to get different dataset splits? 

I guess setting the torch seed might fix this? However I am concerned setting it might affect other functions such as `negative_sampling` which would be better if it remained random.

**Comment by rusty1s** [comment]
You may want to call this function via:
```python
torch.manual_seed(12345)
train_test_split_edges()
torch.manual_seed(time())
```
but I'm actually not sure if that's the best way how you want to do it in PyTorch.

**Reply by sbonner0** [reply]
Great thanks - maybe adding a seed argument for this function might be something that would be useful? Having some control over the dataset splits would be really great.

**Reply by rusty1s** [reply]
I'm not sure. PyTorch does not seem to utilize a `seed` argument anywhere in its codebase (please correct me if I'm wrong), and I generally try match the design principles of PyTorch quite closely. IMO, if you want controllable and deterministic experiments, it's best to set a global seed at the start of your script.

**Reply by sbonner0** [reply]
Perfectly valid thanks! 

Am I right in assuming that the `negative_sampling` function would also be affected by setting the seed? This might not be desirable for training as it would be better to have a different set of negatives for each epoch.

**Reply by rusty1s** [reply]
You will still have a different set of negatives for each epoch, as long as you do not re-set the seed every time before calling `negative_sampling`.
